lib/invitations/inviteUserByEmail.ts

```diff
@@ -0,0 +1,208 @@
+import { FormRecord, UserAbility } from "@lib/types";
+import { getUser } from "@lib/users";
+import {
+  InvalidDomainError,
+  MismatchedEmailDomainError,
+  TemplateNotFoundError,
+  UserAlreadyHasAccessError,
+  UserNotFoundError,
+} from "./exceptions";
+import { getTemplateWithAssociatedUsers } from "@lib/templates";
+import { prisma } from "@lib/integration/prismaConnector";
+import { sendEmail } from "@lib/integration/notifyConnector";
+import { inviteToCollaborate } from "@lib/invitations/emailTemplates/inviteToCollaborate";
+import { inviteToForms } from "@lib/invitations/emailTemplates/inviteToForms";
+import { getOrigin } from "@lib/origin";
+import { logMessage } from "@lib/logger";
+import { Invitation } from "@prisma/client";
+import { logEvent } from "@lib/auditLogs";
+import { isValidGovEmail } from "@lib/validation/validation";
+
+/**
+ * Invite someone to the form by email
+ *
+ * @param ability
+ * @param email
+ * @param formId
+ */
+export const inviteUserByEmail = async (
+  ability: UserAbility,
+  email: string,
+  formId: string,
+  message: string
+) => {
+  let invitation;
+
+  const sender = await getUser(ability, ability.userID);
+
+  if (!sender) {
+    throw new UserNotFoundError();
+  }
+
+  const template = await getTemplateWithAssociatedUsers(ability, formId);
```

Comment:
I'm pretty sure the `getTemplateWithAssociatedUsers()` requires global User and Manage All Forms permissions.  If this function will be available to normal base users this may throw an `AccessControlError`.

Reply:
Updated in the second PR: https://github.com/cds-snc/platform-forms-client/pull/4392/files#diff-c0ef026f39b595cdd15c9a54804ed75e8e473ebc66487370e63ab189cfc4cffaR485-L483